Add tests and update Api wrapper

DerYannis1 opened the pull request:
add more tests and updated some functions of the api wrapper
closes #13 
Enhancements regarding #5

Commit:
save working state

Commit:
tests are working properly, manage external groups still is still a todo

Commit:
external groups are added as well and updated to api/v2, done with the tests

v411e commented on t/tests.py:
```diff
@@ -10,50 +10,116 @@
-    def testGetServerInfo(self):
-        self.assertEqual(api.get_server_info()["restPort"], 11401)
+    """Managing Backup Tasks"""
+
+    
+
```
Please consider using a formatter like [Ruff](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff) for auto-formatting. These line breaks are huuge :)

v411e reviewed (changes requested):
Looks good, thanks a lot! I'd be glad if you could auto-format it before we merge it

DerYannis1 committed:
fixed tests for external_groups

DerYannis1 committed:
Updated external_group_find

DerYannis1 committed:
fixed create_backuptask and update_client

v411e commented on src/sepsesam/api.py:
```diff
@@ -498,7 +498,7 @@ def acl_find(self, **kwargs):
         url = self._urlexpand(endpoint)
         data = {}
         if "queryMode" in kwargs:
```
Is `queryMode` deprecated? If so, we can also delete these lines

DerYannis1 replied:
It is deprecated and can be deleted.
```suggestion
       
```

v411e commented on src/sepsesam/api.py:
```diff
@@ -1507,21 +1510,53 @@ def external_group_get(self, id):
+        
         """
-        return self._filter(self.external_group_list(), **kwargs)
+        log.debug("Running function")
```
This log is not very expressive

```suggestion
```

DerYannis1 committed:
Update src/sepsesam/api.py

remove unnecessary log

Co-authored-by: Valentin Rieß <valentin.riess@posteo.de>

DerYannis1 committed:
removed queryMode at acl_find method

DerYannis1 committed:
Added more tests

Tests for drive_groups, schedule, command and command_event.

DerYannis1 committed:
Updated some methods to API V2

Updated command, command_event, schedule and drive_group to API V2

- command_event_find is still not working

DerYannis1 committed:
fixed tests for schedule methods

v411e commented on src/sepsesam/api.py:
```diff
@@ -1836,79 +1853,105 @@ def command_event_get(self, id):
         """
         Get a drive group
         """
+        if name:
+            id = self.drive_group_resolveDriveGroupToId(name)
+
```
Awesome!

v411e commented on t/tests.py:
```diff
@@ -118,6 +118,125 @@ def test_LocationsCreateAndDelete(self):
+        #creating second event for find() testing
+        api.command_event_create(**{ "id": 1, "name": "testEvent2", "commandName":"eventCommand1", "clientId": 0})
+        #command_event_find not working yet
+        #self.assertTrue(len(api.command_event_find(**{"commandName": "eventCommand1"}))>1)
```
Does this work now? :)

v411e commented on t/tests.py:
```diff
@@ -118,6 +118,125 @@ def test_LocationsCreateAndDelete(self):
+        self.assertEqual(api.command_event_get(1)["object"], "eventCommand2")
+
+        #command_event_find not working yet
+        #len(api.command_event_find(**{"commandName": "eventCommand1"}))==1
```
Does this work now?

DerYannis1 committed:
Added command_event_find tests

DerYannis1 committed:
added/changed  some comments
